mirrord-layer/src/files.rs

```diff
@@ -107,7 +113,28 @@ pub(crate) fn open(path: PathBuf, open_options: OpenOptionsInternal) -> Result<R
     let open_file = file_channel_rx.blocking_recv()?;
 
     let fake_file_name = CString::new(open_file.fd.to_string()).unwrap();
-    let local_file_fd = unsafe { libc::memfd_create(fake_file_name.as_ptr(), 0) };
+
+    let mode = (open_flags & O_RDONLY) | (open_flags | O_RDONLY) | O_CREAT;
+    // TODO(alex) [mid] 2022-05-24: Be very careful here, if a call to create the local fd fails,
+    // the remote one stays up (basically a leak). So I need a way to `close` the remote on failure
+    // here.
+    //
+    // NOTE(alex): The pair `shm_open`, `shm_unlink` are used to create a temporary file
```

Comment:
why do we care about flags though? we "tunnel" those flags to the agent and we hook all operations on the file (presumably) so the fact it's shm or memfd doesn't really matter?

Reply:
Yeah, I'm just being cautious here, hoping to mimic what the file system would expect as much as possible.